src/places/places.service.ts

```diff
@@ -38,77 +44,92 @@ export class PlacesService extends createPrismaBase(MODELS.Place) {
       ...(slug ? { slug } : {}),
     }
 
-    let places: (
-      | Prisma.PlaceGetPayload<{ select: Prisma.PlaceSelect }>
-      | Prisma.PlaceGetPayload<{ include: Prisma.PlaceInclude }>
-    )[]
+    const placeSelectBase: Prisma.PlaceSelect | undefined = placeColumns
+      ? (buildColumnSelect(placeColumns) as Prisma.PlaceSelect)
+      : undefined
 
     const buildRaceInclude = () => {
-      if (raceColumns) {
-        const raceSelect = buildColumnSelect(raceColumns) as Prisma.RaceSelect
-        return { select: raceSelect }
-      } else {
-        return true
+      if (!raceColumns) return true
+      if (!includeRaces) return true
+
+      // Force add slug and positionNames so we can dedupe by slug
+      const cols = Array.from(
+        new Set([
+          ...raceColumns.split(','),
+          SLUG_COLUMN_NAME,
+          POSITION_NAMES_COLUMN_NAME,
+        ]),
+      ).join(',')
+
+      return {
+        select: buildColumnSelect(cols) as Prisma.RaceSelect,
       }
     }
```

Comment:
Don't write methods within methods.  Based on the purpose of the method, figure out where it should live, and how it should be accessed and by what.

Also, don't use closure to control logic flow. If race columns and include races are two dependent values for how to build race include will flow, then that should be apparent in the function signature and they should be passed in as arguments. Just as an example.